src/patternfly/components/Button/button.scss

```diff
@@ -143,6 +143,12 @@
   --#{$button}--m-plain--active--Color: var(--#{$pf-global}--Color--100);
   --#{$button}--m-plain--disabled--Color: var(--#{$pf-global}--disabled-color--200);
   --#{$button}--m-plain--disabled--BackgroundColor: transparent;
+  --#{$button}--m-plain--m-inline--FontSize: inherit;
```

Comment:
`inherit` (and `unset` and `initial`) can't be used as the value of a CSS var, and have that value come out of the `var()` function that refers to it. You'll need to use `inherit` as the fallback to a CSS var in the `var()` function. From there you can trigger the fallback by setting the var's value to `initial`. Technically `inherit` will trigger the fallback if one is set, too, but `initial` is the designated value to always trigger the fallback. 
You can read about that here - https://www.w3.org/TR/css-variables-1/#example-ee865e1b

Your PR is working because in `font-size: var(--#{$button}--m-plain--m-inline--FontSize);`, the custom property in the `var()` function is invalid, so `var()` is returning `initial` as the value... and the `initial` value for `font-size` is `inherit`.